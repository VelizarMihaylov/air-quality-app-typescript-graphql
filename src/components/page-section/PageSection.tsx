import React from 'react'
import PropTypes from 'prop-types'

import styled from 'styled-components'

import { ContentWrapper } from 'components/content-wrapper'

type PageSectionProps = {
  children?: React.ReactNode
}

const StyledPageSection = styled.section`
  margin-bottom: ${({ theme }) => theme.spacing(1.5)};
`

export const PageSection: React.FC<PageSectionProps> = ({ children }) => {
  return (
    <StyledPageSection>
      <ContentWrapper>{children}</ContentWrapper>
    </StyledPageSection>
  )
}

PageSection.propTypes = {
  children: PropTypes.node
}
