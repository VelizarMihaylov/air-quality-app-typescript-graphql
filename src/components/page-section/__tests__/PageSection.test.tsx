import React from 'react'
import { create } from 'react-test-renderer'

import { PageSection } from '../PageSection'

import { ThemeProvider } from 'styled-components'
import generateTheme from 'theme'

describe('PageSection', () => {
  it('should render without throwing an error', () => {
    const renderer = create(
      <ThemeProvider theme={generateTheme()}>
        <PageSection />
      </ThemeProvider>
    )

    expect(renderer.toJSON()).toMatchSnapshot()
  })
})
