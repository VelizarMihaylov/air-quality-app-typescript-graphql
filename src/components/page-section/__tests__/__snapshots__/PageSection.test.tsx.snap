// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageSection should render without throwing an error 1`] = `
<section
  className="sc-gsTCUz dtsAoy"
>
  <div
    className="sc-bdfBwQ gTLSEq"
  />
</section>
`;
